src/main/java/org/candlepin/insights/inventory/kafka/KafkaEnabledInventoryService.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2009 - 2019 Red Hat, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * Red Hat trademarks are not licensed under GPLv3. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.insights.inventory.kafka;
+
+import org.candlepin.insights.inventory.ConduitFacts;
+import org.candlepin.insights.inventory.DefaultInventoryService;
+import org.candlepin.insights.inventory.client.InventoryServiceProperties;
+import org.candlepin.insights.inventory.client.resources.HostsApi;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.kafka.core.KafkaTemplate;
+
+import java.time.OffsetDateTime;
+import java.util.List;
+
+/**
+ * An InventoryService implementation that includes a Kafka producer that is capable
+ * of sending messages to the inventory service's Kafka instance.
+ */
+public class KafkaEnabledInventoryService extends DefaultInventoryService {
+
+    private static final Logger log = LoggerFactory.getLogger(KafkaEnabledInventoryService.class);
+
+    private final KafkaTemplate<String, HostOperationMessage> producer;
+    private final String hostIngressTopic;
+
+    public KafkaEnabledInventoryService(InventoryServiceProperties inventoryServiceProperties,
+        KafkaTemplate<String, HostOperationMessage> producer, HostsApi hostsInventoryApi) {
+        super(hostsInventoryApi);
+        this.producer = producer;
+        this.hostIngressTopic = inventoryServiceProperties.getKafkaHostIngressTopic();
+    }
+
+    @Override
+    public void sendHostUpdate(List<ConduitFacts> facts) {
+        if (facts.isEmpty()) {
+            log.info("No facts to report!");
+            return;
+        }
+
+        OffsetDateTime now = OffsetDateTime.now();
+        for (ConduitFacts factSet : facts) {
+            log.debug("Sending: {}:{}:{}", factSet.getAccountNumber(), factSet.getOrgId(),
+                factSet.getSubscriptionManagerId());
+
+            // Attempt to send the host create/update message. If the send fails for any reason,
```

Comment:
We might want to retry here in case of transient Kafka failures (e.g. short outage). We could use spring retry like #102. If you find the idea agreeable, we can make a separate card for it.

Reply:
:+1:  Sounds like a good idea. I'll make a card for it.

Reply:
https://projects.engineering.redhat.com/browse/ENT-1573